node/network/protocol/src/lib.rs

```diff
@@ -184,14 +185,69 @@ macro_rules! our_view {
 /// and the highest known finalized block number.
 ///
 /// Up to `N` (5?) chain heads.
-#[derive(Default, Debug, Clone, PartialEq, Eq, Encode, Decode)]
+#[derive(Default, Debug, Clone, Encode, Decode)]
 pub struct View {
 	/// A bounded amount of chain heads.
+	/// Invariant: Contains no duplicates.
 	pub heads: Vec<Hash>,
 	/// The highest known finalized block number.
 	pub finalized_number: BlockNumber,
 }
 
+impl View {
+	/// Checks if one view is a superset of another.
+	///
+	/// Generally this only makes sense for match
+	/// `finalized_number`.
+	/// Equivalence is also.
+	pub fn is_superset_of(&self, other: &Self) -> bool {
+		// None of the others' heads must be missed in our heads.
+		// Resistant to duplicates.
+		other.heads.iter().filter(|x| !self.heads.contains(x)).count() == 0
+	}
+}
+
+impl PartialEq for View {
```

Comment:
I don't see any advantage of this self written `PartialEq` implementation.

Reply:
That's correct, given the assumptions there are no duplicates. To be removed.